packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenPaymentRolesWithManager.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../../../access-control-registry/RoleDeriver.sol";
+import "../../../access-control-registry/AccessControlClient.sol";
+import "../../../access-control-registry/interfaces/IAccessControlRegistry.sol";
+import "./interfaces/IAirnodeTokenPaymentRolesWithManager.sol";
+
+/// @title Contract that implements generic AccessControlRegistry roles for
+/// the AirnodePaymentLock contract
+contract AirnodeTokenPaymentRolesWithManager is
+    RoleDeriver,
+    AccessControlClient,
+    IAirnodeTokenPaymentRolesWithManager
+{
+    // There are 3 roles implemented in this contract:
+    // Root
+    // └── (1) Admin (can grant and revoke the roles below)
+    //     ├── (2) Payment token price setter (Oracle)
+    //     ├── (3) Airnode to minimum whitelist duration setter
```

Comment:
can probably change this to whitelist duration setter for both minimum and maximum if we do end up setting both

Reply:
This might actually be a good idea to avoid creating a new role 👍🏻 if we do add the setter.